framework/source/class/qx/lang/String.js

```diff
@@ -299,7 +299,7 @@ qx.Bootstrap.define("qx.lang.String",
     format : function(pattern, args)
     {
       var str = pattern;
-      var regexp = /%(\d+)|%{(\S*)}/g;
+      var regexp = /%(\d+)|%{(\S[^{]*)}/g;
```

Comment:
I think the added portion should be `[^}]` instead of `[^{]` shouldn't it? We want to match everything up to the closing brace. Regardless of the use case (consecutive replacements as described in this PR or just a single replacement), a closing brace is not a legal character in the replacement variable name.

Reply:
An interesting question is whether a space is a legal character in the replacement variable name. Previously it was not allowed. With this change, it would be. I suspect that's incorrect; spaces should not be allowed in the variable name.

Reply:
You're right it's clearly an error, i made differents tests and finally took the bad version of the regex. Thank you for notice

Reply:
clearly a good question. I agree with you however I don't know how to do the regex to solve the problem and avoir the whitespace